web/src/shared/components/Graphs/StackedBarChart.tsx

```diff
@@ -0,0 +1,384 @@
+import * as d3 from 'd3'
+import React from 'react'
+
+export interface IStackedBarChartData {
+    date: Date
+    values: { [key: string]: number }
+}
+
+interface IStackedBarChartProps {
+    data?: IStackedBarChartData[]
+    accumulate: boolean
+}
+
+function getSeries(data: IStackedBarChartData[] | undefined) {
+    if (!data || data.length === 0) {
+        return []
+    }
+
+    return Object.keys(data[0].values)
+}
+
+function alignToStartOfMonth(date: Date): Date {
+    const year = date.getFullYear()
+    const month = date.getMonth()
+    return new Date(Date.UTC(year, month, 1))
+}
+
+function createNewDates(lastDate: Date, differenceInDays: number): Date[] {
+    const newDates: Date[] = []
+    for (let i = 1; i <= 3; i++) {
+        const newDate = new Date(lastDate.getTime() + i * differenceInDays * 24 * 60 * 60 * 1000)
```

Comment:
If the difference is 29 days, do you potentially create two entries on the same date at the start of the month?

Reply:
We are only predicting 3 records so it would not, it could be potential problem if something like 100 extra records would need to be predicted.
Basically we look at the last 2 record time difference and if it is more than 28 days we are generating additional months, not days or weeks.